### PR TITLE
Max Subarray Sum Java Implementation of Kadane's Algorithm

### PR DESCRIPTION
**Pull request by brad182**
added java implementation

### CODE REVIEW

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 1]
```diff
@@ -172,6 +172,46 @@ int main() {
...
+			/*
+			Continue the subarray sum or start a new
+			subarray sum beginning at the current element.
+			*/
+			currentSum = Math.max(currentSum + x[i], x[i]);
+			// Store the maximum subarray sum at each iteration.
+			maxSubarraySum = Math.max(maxSubarraySum, currentSum);
+		}
+
+		System.out.println(maxSubarraySum);
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:207]:
```suggestion
		io.println(maxSubarraySum);
		io.close();
```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 2]
```diff
@@ -172,6 +172,46 @@ int main() {
...
+		}
+
+		long currentSum = x[0];
+		long maxSubarraySum = x[0];
+
+		for (int i = 1; i < n; i++) {
+			/*
+			Continue the subarray sum or start a new
+			subarray sum beginning at the current element.
+			*/
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:201]:
```suggestion
			/*
			* Continue the subarray sum or start a new
			* subarray sum beginning at the current element.
			*/
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 3]
```diff
@@ -172,6 +173,47 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+### Java Implementation
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:179]:
```suggestion
## Java Implementation

```

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 4]
```diff
@@ -172,6 +173,47 @@ int main() {
...
+```java
+
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:181]:
```suggestion
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 5]
```diff
@@ -69,7 +69,8 @@ int main(){
 
 </CPPSection>
 
-<JavaSection>
+<
+	Section>
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:73]: ??

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 6]
```diff
@@ -69,7 +69,7 @@ int main(){
 
 </CPPSection>
 
-<JavaSection>
+<Section>
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:72]:
```suggestion
<JavaSection>
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 7]
```diff
@@ -172,6 +172,47 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+	
+## Java Implementation
```
Comment (HarryWangATX) [inline content/3_Silver/Prefix_Sums_2.mdx:178]:
```suggestion
### Java Implementation
```

mb there should be 3 #'s

**Review by HarryWangATX — APPROVED** [review]
👍👍

**Review by dongliu0426 — APPROVED** [review]

File: content/3_Silver/Prefix_Sums_2.mdx [hunk 8]
```diff
@@ -172,6 +172,47 @@ int main() {
...
 </CPPSection>
+	
+<JavaSection>
+	
+### Java Implementation
+
+```java
+import java.io.*;
+import java.util.*;
+public class maxSubarraySumKadane {
```
Comment (SansPapyrus683) [inline content/3_Silver/Prefix_Sums_2.mdx:183]: Add whitespace between lines 182 and 183, and change the name of the class so it's PascalCase please.

**Review by SansPapyrus683 — APPROVED** [review]
yeah that's it, looks good to me

**Pull request merged**